doc/api/addons.markdown

```diff
@@ -1,39 +1,47 @@
 # Addons
 
-Addons are dynamically-linked shared objects. They can provide glue to C and
-C++ libraries. The API (at the moment) is rather complex, involving
-knowledge of several libraries:
+Node.js Addons are dynamically-linked shared objects, written in C or C++, that
+can be loaded into Node.js using the `require()` statement, and used just as if
+they were an ordinary Node.js module. They are used primarily to provide an
+interface between JavaScript running in Node.js and C/C++ libraries.
 
- - V8 JavaScript, a C++ library. Used for interfacing with JavaScript:
-   creating objects, calling functions, etc.  Documented mostly in the
+At the moment, The method for implementing Addons is rather complicated,
+involving knowledge of several components and APIs :
+
+ - V8 JavaScript: the C++ library Node.js currently uses to provide the
+   JavaScript implementation. V8 provides the mechanisms for creating objects,
+   calling functions, etc. V8's API is documented mostly in the
    `v8.h` header file (`deps/v8/include/v8.h` in the Node.js source
    tree), which is also available [online][].
 
- - [libuv][], C event loop library. Anytime one needs to wait for a file
-   descriptor to become readable, wait for a timer, or wait for a signal
-   to be received, one will need to interface with libuv. That is, if you
-   perform any I/O, libuv will need to be used.
-
- - Internal Node.js libraries. The most important class is `node::ObjectWrap`
-   which you will likely want to derive from.
+ - [libuv][]: The C library that implements the Node.js event loop. Anytime
+   an application needs to wait for a file descriptor to become readable, wait
+   for a timer, or wait for a signal to be received, the application will need
+   to interface with libuv. That is, libuv is required any time I/O operations
+   are used.
 
- - Others. Look in `deps/` for what else is available.
+ - Internal Node.js libraries. Node.js itself exports a number of C/C++ APIs
+   that Addons can use -- the most important of which is the `node::ObjectWrap`
+   class.
 
-Node.js statically compiles all its dependencies into the executable.
-When compiling your module, you don't need to worry about linking to
-any of these libraries.
+ - Node.js includes a number of others statically linked libraries including
+   c-ares, OpenSSL, zlib and http_parser. Each of these may be used to various
+   extents by Addons. Look in the `deps/` folder in the Node.js source tree to
+   see what else is available. (Note that Node.js statically compiles all its
```

Comment:
There's a tricky additional piece of information here that we may or may not want to include. You don't actually get your dependencies in the same form as `deps/` in most cases, node-gyp may download the full source tarball or you may `--nodedir` compile it in which case you _do_ but it may also just be downloading the headers and get a proper `include/` directory. Mostly this just works and you just `#include <whatever>` and it finds it, see https://github.com/nodejs/node-gyp/blob/master/addon.gypi#L6-L12 for the details of how these various options work together. Pointing people to `deps/` is misleading because `gtest` is in there (we need to move this to `tools/`) and `http_parser` is also there but not exposed to addons afaik.

Sorry, not sure how to word all of this, just that I don't think it's as simple as "Look in the `deps/` folder".